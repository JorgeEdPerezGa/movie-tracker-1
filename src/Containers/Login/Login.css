.welcome-text {
  font-size: 32px;
  margin-top: 0;
  font-weight: 100; }

.login {
  background-color: #ececec;
  height: 100vh;
  width: 60%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  align-content: space-around;
  margin-left: auto;
  margin-right: auto; }

.email, .password {
  width: 90%;
  margin-bottom: 23px;
  height: 35px;
  font-size: 17px; }

.submit-button {
  width: 92%;
  height: 40px;
  font-size: 18px;
  color: white;
  background: #5600ff;
  border: none;
  font-weight: 100; }
  .submit-button:hover {
    cursor: pointer; }
  .submit-button:focus {
    outline: 0 !important; }
