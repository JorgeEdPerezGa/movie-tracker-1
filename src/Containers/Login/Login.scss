.login {
  background-color: black;
  height: 100vh;
  width: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: space-around;
  margin-left: auto;
  margin-right: auto;
}

.email, .password {
  width: 120px;
}

.submit-button{
  width: 80%;
  height: 40px;
}
