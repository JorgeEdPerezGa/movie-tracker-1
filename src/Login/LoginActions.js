import { userLogin }  from '../API/helper'

export const loginAction = (user) => {
  return async (dispatch) => {
     const callUser = await userLogin(user);
     console.log(callUser)
     return dispatch(loginSuccess(callUser.data));
  }
}

export const loginSuccess = (user) => {
  return {
    type: 'LOGIN_SUCCESS',
    user
  }
}
