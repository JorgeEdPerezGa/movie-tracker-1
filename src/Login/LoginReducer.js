export const loginReducer = (store = [], action) => {
  switch(action.type) {
    case 'LOGIN_USER':
    return [...store, action.user]
    case 'LOGIN_SUCCESS':
    return [...store, action.user];
    default:
    return store;
  }
}
