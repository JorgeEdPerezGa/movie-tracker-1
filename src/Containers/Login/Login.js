import React, {Component} from 'react';
import { connect } from 'react-redux';
import { loginSuccess } from '../../Actions/Actions';
import helper from '../../API/helper.js';
import { browserHistory } from 'react-router-dom'
import './Login.css'

const { userLogin } = helper;



class Login extends Component {
  constructor() {
    super()
    this.state = {
      email: 'jason@email.com',
      password: 'password',
      loginError: false
    }
  }

handleChange = (key, event) => {
  this.setState({[key]: event.target.value})
}

logIn = async (cred) => {
  const loggedIn = await userLogin(cred);

  if (loggedIn) {
    this.props.loginSuccess(loggedIn);
    this.props.history.push('/')
  } else {
    alert('invalid login');
  }
}


  render() {
    return(
      <div className = 'login'>
        <h2 className='welcome-text'>Welcome Back!</h2>
        <input
          className = 'email'
          placeholder = 'email'
                 value={this.state.email}
              onChange={(event) => this.handleChange('email', event)}
        />
        <input
          className = 'password'
          placeholder = 'password'
                 value={this.state.passord}
              onChange={(event) => this.handleChange('password', event)}
        />
        <button
          className = 'submit-button'
          onClick={(event) => this.logIn(this.state)}>submit</button>
      </div>
    )
  }
}

const mapStateToProps = (store) => {
  return {
    login: store.login
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    loginSuccess: (email, password) => {
      dispatch(loginSuccess(email, password))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);
