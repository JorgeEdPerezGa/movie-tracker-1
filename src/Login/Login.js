import React, {Component} from 'react';
import { connect } from 'react-redux';
import { loginAction } from './LoginActions.js';
import helper from '../API/helper.js';
const { userLogin } = helper;


class Login extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: ''
    }
  }

handleChange = (key, event) => {
  this.setState({[key]: event.target.value})
}



  render() {
    return(
      <div className = 'login'>
        <h2>LOGIN PAGE</h2>
        <input
          placeholder = 'email'
                 value={this.state.email}
              onChange={(event) => this.handleChange('email', event)}
        />
        <input
          placeholder = 'password'
                 value={this.state.passord}
              onChange={(event) => this.handleChange('password', event)}
        />
        <button onClick={(event) => this.props.userLogin(this.state)}>submit</button>
      </div>
    )
  }
}

const mapStateToProps = (store) => {
  return {
    login: store.login
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    userLogin: (email, password) => {
      dispatch(loginAction(email, password))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);
