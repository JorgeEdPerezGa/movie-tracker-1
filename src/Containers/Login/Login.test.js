import React from 'react';
import { shallow, mount } from 'enzyme';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';

describe('Login Conatiner', () => {
  describe('login', () => {
    let login;

    beforeEach( () => {
  login = shallow(
    <Login
      name='Jorge'
    />);
  });

  it('should exist', () => {
    expect(login).toBeDefined();
  });

  it('should match the snapshot', () => {
  expect(login).toMatchSnapshot();
  });

  it('should have a default state', () => {
    const expectedDefault = {
      "email": "",
      "loginError": false,
      "password": ""}
    expect(login.state()).toEqual(expectedDefault);
  })
  })

  describe('mapStateToProps', () => {
    let mockStore;
    let result;

    beforeAll(() => {
      mockStore = {
        login: undefined,
        user: {
          info: {
            name: "Jorge"
          }
        },
      };
      result = mapStateToProps(mockStore);
    });

    it('should pull user name from the store', () => {
      expect(result.user.info.name).toEqual(mockStore.user.info.name);
    });
  })

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let result;

    beforeEach(() => {
      mockDispatch = jest.fn();
      result = mapDispatchToProps(mockDispatch);
    });

    it('should call dispatch when loginSuccess is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.loginSuccess(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call dispatch when setFavorites is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.setFavorites(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
})
