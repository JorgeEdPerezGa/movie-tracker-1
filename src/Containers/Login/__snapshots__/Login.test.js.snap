// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Conatiner login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    name="Jorge"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="welcome-text"
        >
          Welcome Back!
        </h2>,
        <input
          className="email"
          onChange={[Function]}
          placeholder="email"
          value=""
        />,
        <input
          className="password"
          onChange={[Function]}
          placeholder="password"
          value={undefined}
        />,
        <button
          className="submit-button"
          onClick={[Function]}
        >
          submit
        </button>,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome Back!",
          "className": "welcome-text",
        },
        "ref": null,
        "rendered": "Welcome Back!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "email",
          "onChange": [Function],
          "placeholder": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "password",
          "onChange": [Function],
          "placeholder": "password",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "submit",
          "className": "submit-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "submit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="welcome-text"
          >
            Welcome Back!
          </h2>,
          <input
            className="email"
            onChange={[Function]}
            placeholder="email"
            value=""
          />,
          <input
            className="password"
            onChange={[Function]}
            placeholder="password"
            value={undefined}
          />,
          <button
            className="submit-button"
            onClick={[Function]}
          >
            submit
          </button>,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome Back!",
            "className": "welcome-text",
          },
          "ref": null,
          "rendered": "Welcome Back!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "email",
            "onChange": [Function],
            "placeholder": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password",
            "onChange": [Function],
            "placeholder": "password",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "submit",
            "className": "submit-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "submit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
